src/daemon/rpc_command_executor.cpp

```diff
@@ -2036,38 +2036,34 @@ static void print_service_node_list_state(cryptonote::network_type nettype, uint
 
     // Print Expiry Info
     {
-      uint64_t expiry_height_auto_stake = entry.registration_height + STAKING_AUTHORIZATION_EXPIRATION_AUTOSTAKE;
-      uint64_t expiry_height            = entry.registration_height;
-
+      uint64_t expiry_height = entry.registration_height;
       if (is_registered) expiry_height += (nettype == cryptonote::TESTNET) ? STAKING_REQUIREMENT_LOCK_BLOCKS_TESTNET : STAKING_REQUIREMENT_LOCK_BLOCKS;
-      else               expiry_height += STAKING_AUTHORIZATION_EXPIRATION_WINDOW;
+      else               expiry_height += (STAKING_AUTHORIZATION_EXPIRATION_WINDOW / DIFFICULTY_TARGET_V2);
 
       if (curr_height)
       {
         uint64_t now = time(nullptr);
-        uint64_t delta_height            = expiry_height - *curr_height;
-        uint64_t delta_height_auto_stake = expiry_height_auto_stake - *curr_height;
+        uint64_t delta_height = expiry_height - *curr_height;
```

Comment:
Why is curr_height a pointer?

Reply:
Optional param, for whatever reason if the RPC call to /getheight fails then we can pass a nullptr and display an alternate message instead.

Reply:
This is following an RPC call to get SN info right?

I'd have just failed out with an error that the daemon wasn't responding

Reply:
(doesn't matter though, it's fine how it is)

Reply:
Having it makes it more resilient than not. We do have all the information they really wanted, curr_height is for auxiliary data like calculating time estimations and displaying how many blocks until X happens.